crates/db_schema/src/impls/federation_allowlist.rs

```diff
@@ -1,93 +1,103 @@
 use crate::{
-  schema::federation_allowlist,
-  source::{
-    federation_allowlist::{FederationAllowList, FederationAllowListForm},
-    instance::Instance,
+  newtypes::InstanceId,
+  schema::{admin_allow_instance, federation_allowlist},
+  source::federation_allowlist::{
+    AdminAllowInstance,
+    AdminAllowInstanceForm,
+    FederationAllowListForm,
   },
   utils::{get_conn, DbPool},
 };
-use diesel::{dsl::insert_into, result::Error};
-use diesel_async::{AsyncPgConnection, RunQueryDsl};
+use diesel::{delete, dsl::insert_into, result::Error, ExpressionMethods, QueryDsl};
+use diesel_async::RunQueryDsl;
 
-impl FederationAllowList {
-  pub async fn replace(pool: &mut DbPool<'_>, list_opt: Option<Vec<String>>) -> Result<(), Error> {
+impl AdminAllowInstance {
```

Comment:
That way the two actions are completely separate, which is more complicated to call. More importantly it doesnt use a transaction so its possible that the mod log entry doesnt get written.

Reply:
We have that issue with all the endpoints and their modlogs. If we add transactions for all these, they should be done at the api action level, then we can do them all at once. 

I'd rather things be uniform, then the cleanup tasks can do them all at once, rather than doing things uniquely each time, then having to go back and fix the other ones one at a time.